fdbserver/storageserver.actor.cpp

```diff
@@ -8442,10 +8455,19 @@ ACTOR Future<Void> updateStorage(StorageServer* data) {
 			    .detail("OldestRemoveKVSRangesVersion", data->pendingRemoveRanges.begin()->first);
 			ASSERT(newOldestVersion <= data->pendingRemoveRanges.begin()->first);
 			if (newOldestVersion == data->pendingRemoveRanges.begin()->first) {
+				state std::vector<std::string> emptyShardIds;
 				for (const auto& range : data->pendingRemoveRanges.begin()->second) {
-					data->storage.removeRange(range);
+					auto ids = data->storage.removeRange(range);
+					emptyShardIds.insert(emptyShardIds.end(), ids.begin(), ids.end());
 					TraceEvent(SevVerbose, "RemoveKVSRange", data->thisServerID).detail("Range", range);
 				}
+				if (emptyShardIds.size() > 0) {
+					state double start = now();
+					TraceEvent("RemoveEmptyPhysicalShards").detail("NumShards", emptyShardIds.size());
+					wait(data->storage.cleanUpShardsIfNeeded(emptyShardIds));
+					data->physicalShardHistograms.cleanUpLatency->sample(now() - start);
```

Comment:
If you are sampling in microseconds, you have to pass microseconds. now() will give the time in seconds I think